cloud-azure/src/main/resources/templates/arm-v2.ftl

```diff
@@ -219,14 +219,19 @@
                         </#list>
                      </#if>
                      },
-                   <#if loadBalancerMapping[instance.groupName]?? && (loadBalancers?filter(loadBalancer -> loadBalancer.sku == "STANDARD")?size > 0)>
+                   <#if instance.instance.availabilityZone?? && instance.instance.availabilityZone?has_content>
+                     "zones": [
+                       "${instance.instance.availabilityZone}"
+                     ],
+                   </#if>
+                   <#if (loadBalancerMapping[instance.groupName]?? && (loadBalancers?filter(loadBalancer -> loadBalancer.sku == "STANDARD")?size > 0)) || (instance.instance.availabilityZone?? && instance.instance.availabilityZone?has_content)>
                      "sku": {
                          "name": "Standard",
                          "tier": "Regional"
                      },
                    </#if>
                    "properties": {
-                       <#if instanceGroup == "GATEWAY" || (instance.availabilitySetName?? && instance.availabilitySetName?has_content)>
+                       <#if multiAz || instanceGroup == "GATEWAY" || (instance.availabilitySetName?? && instance.availabilitySetName?has_content)>
```

Comment:
We should verify this condition, my guess is that only the `GATEWAY` instance group should have `Static` public ip allocation method by default.